pkg/apis/componentconfig/types.go

```diff
@@ -150,6 +150,9 @@ type KubeletConfiguration struct {
 	// default /var/run/kubernetes). If tlsCertFile and tlsPrivateKeyFile
 	// are provided, this flag will be ignored.
 	CertDirectory string `json:"certDirectory"`
+	// BootstrapKubeConfig is the path to config file or directory of files that
+	// will be used to get client certificate for kubelet.
+	BootstrapKubeconfig string `json:"bootstrapKubeconfig"`
```

Comment:
Oops it's not even in KubeletConfig, it's in KubeProxy config. But

We should divide  config into three buckets:
* How do I find a trust the apiserver
* How does the apiserver trust me
* Everything else
These configs have orthogonal concerns,  different writer and should be sourced from three different files.  They should have no dependencies. Are you planning on merging KubeletConfig from mulitple files and from the API? Please say no.

All components intentionally do not include Kubeconfig in their componentconfig:

https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/options/options.go#L48
https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-controller-manager/app/options/options.go#L39
https://github.com/kubernetes/kubernetes/blob/master/plugin/cmd/kube-scheduler/app/options/options.go#L35
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go#L45

I'm not sure what the KubeconfigPath is doing in kube-proxy.... It seems to be unused. So as not to derail this discussion, let's move to https://github.com/kubernetes/kubernetes/issues/30805